openwisp_firmware_upgrader/api/serializers.py

```diff
@@ -45,10 +46,31 @@ class Meta:
 
 
 class FirmwareImageSerializer(BaseSerializer):
+    file = serializers.FileField(required=False)
+    file_url = serializers.SerializerMethodField(read_only=True)
+
     def validate(self, data):
         data['build'] = self.context['view'].get_parent_queryset().get()
         return super().validate(data)
 
+    def get_file_url(self, obj):
+        request = self.context.get('request')
+        if request and getattr(obj, 'pk', None):
+            return request.build_absolute_uri(
+                reverse(
+                    'upgrader:api_firmware_download',
+                    args=[obj.build.pk, obj.pk],
+                )
+            )
+        return obj.file.url if hasattr(obj, 'file') else None
+
+    def to_representation(self, instance):
+        ret = super().to_representation(instance)
+        # Replace 'file' with the URL from file_url
+        if 'file_url' in ret:
+            ret['file'] = ret.pop('file_url')
+        return ret
```

Comment:
Since we are already overriding the value of `field` value in the to_representation, then why do we need to add another field `file_url` to the serializer? We can directly call the `get_file_url`